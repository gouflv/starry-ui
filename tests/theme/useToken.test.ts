import { mount } from '@vue/test-utils'
import { expect, test } from 'vitest'
import TokenRender from './components/TokenRender.vue'

test('should TokenRender works', () => {
  const vm = mount(TokenRender)
  expect(vm.html()).toMatchSnapshot('all tokens')
})
