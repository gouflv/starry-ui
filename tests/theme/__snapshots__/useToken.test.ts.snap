// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should TokenRender works > all tokens 1`] = `
"<div><span>blue:#3a8bff</span><span>purple:#722ED1</span><span>cyan:#13C2C2</span><span>green:#52C41A</span><span>magenta:#EB2F96</span><span>pink:#eb2f96</span><span>red:#F5222D</span><span>orange:#FA8C16</span><span>yellow:#FADB14</span><span>volcano:#FA541C</span><span>geekblue:#2F54EB</span><span>gold:#FAAD14</span><span>lime:#A0D911</span><span>colorPrimary:#3a8bff</span><span>colorSuccess:#52c41a</span><span>colorWarning:#faad14</span><span>colorError:#ff4d4f</span><span>colorInfo:#3a8bff</span><span>colorTextBase:</span><span>colorBgBase:</span><span>fontFamily:-apple-system, BlinkMacSystemFont, 'Segoe UI', Roboto, 'Helvetica Neue', Arial,
'Noto Sans', sans-serif, 'Apple Color Emoji', 'Segoe UI Emoji', 'Segoe UI Symbol',
'Noto Color Emoji'</span><span>fontSize:14</span><span>lineWidth:1</span><span>lineType:solid</span><span>borderRadius:6</span><span>sizeUnit:4</span><span>sizeStep:4</span><span>sizePopupArrow:16</span><span>controlHeight:32</span><span>zIndexBase:0</span><span>zIndexPopupBase:1000</span><span>motionUnit:0.1</span><span>motionBase:0</span><span>blue-1:#f0f9ff</span><span>blue-2:#def0ff</span><span>blue-3:#b5dbff</span><span>blue-4:#8cc4ff</span><span>blue-5:#63a9ff</span><span>blue-6:#3a8bff</span><span>blue-7:#276bd9</span><span>blue-8:#174eb3</span><span>blue-9:#0b348c</span><span>blue-10:#072266</span><span>purple-1:#f9f0ff</span><span>purple-2:#efdbff</span><span>purple-3:#d3adf7</span><span>purple-4:#b37feb</span><span>purple-5:#9254de</span><span>purple-6:#722ed1</span><span>purple-7:#531dab</span><span>purple-8:#391085</span><span>purple-9:#22075e</span><span>purple-10:#120338</span><span>cyan-1:#e6fffb</span><span>cyan-2:#b5f5ec</span><span>cyan-3:#87e8de</span><span>cyan-4:#5cdbd3</span><span>cyan-5:#36cfc9</span><span>cyan-6:#13c2c2</span><span>cyan-7:#08979c</span><span>cyan-8:#006d75</span><span>cyan-9:#00474f</span><span>cyan-10:#002329</span><span>green-1:#f6ffed</span><span>green-2:#d9f7be</span><span>green-3:#b7eb8f</span><span>green-4:#95de64</span><span>green-5:#73d13d</span><span>green-6:#52c41a</span><span>green-7:#389e0d</span><span>green-8:#237804</span><span>green-9:#135200</span><span>green-10:#092b00</span><span>magenta-1:#fff0f6</span><span>magenta-2:#ffd6e7</span><span>magenta-3:#ffadd2</span><span>magenta-4:#ff85c0</span><span>magenta-5:#f759ab</span><span>magenta-6:#eb2f96</span><span>magenta-7:#c41d7f</span><span>magenta-8:#9e1068</span><span>magenta-9:#780650</span><span>magenta-10:#520339</span><span>pink-1:#fff0f6</span><span>pink-2:#ffd6e7</span><span>pink-3:#ffadd2</span><span>pink-4:#ff85c0</span><span>pink-5:#f759ab</span><span>pink-6:#eb2f96</span><span>pink-7:#c41d7f</span><span>pink-8:#9e1068</span><span>pink-9:#780650</span><span>pink-10:#520339</span><span>red-1:#fff1f0</span><span>red-2:#ffccc7</span><span>red-3:#ffa39e</span><span>red-4:#ff7875</span><span>red-5:#ff4d4f</span><span>red-6:#f5222d</span><span>red-7:#cf1322</span><span>red-8:#a8071a</span><span>red-9:#820014</span><span>red-10:#5c0011</span><span>orange-1:#fff7e6</span><span>orange-2:#ffe7ba</span><span>orange-3:#ffd591</span><span>orange-4:#ffc069</span><span>orange-5:#ffa940</span><span>orange-6:#fa8c16</span><span>orange-7:#d46b08</span><span>orange-8:#ad4e00</span><span>orange-9:#873800</span><span>orange-10:#612500</span><span>yellow-1:#feffe6</span><span>yellow-2:#ffffb8</span><span>yellow-3:#fffb8f</span><span>yellow-4:#fff566</span><span>yellow-5:#ffec3d</span><span>yellow-6:#fadb14</span><span>yellow-7:#d4b106</span><span>yellow-8:#ad8b00</span><span>yellow-9:#876800</span><span>yellow-10:#614700</span><span>volcano-1:#fff2e8</span><span>volcano-2:#ffd8bf</span><span>volcano-3:#ffbb96</span><span>volcano-4:#ff9c6e</span><span>volcano-5:#ff7a45</span><span>volcano-6:#fa541c</span><span>volcano-7:#d4380d</span><span>volcano-8:#ad2102</span><span>volcano-9:#871400</span><span>volcano-10:#610b00</span><span>geekblue-1:#f0f5ff</span><span>geekblue-2:#d6e4ff</span><span>geekblue-3:#adc6ff</span><span>geekblue-4:#85a5ff</span><span>geekblue-5:#597ef7</span><span>geekblue-6:#2f54eb</span><span>geekblue-7:#1d39c4</span><span>geekblue-8:#10239e</span><span>geekblue-9:#061178</span><span>geekblue-10:#030852</span><span>gold-1:#fffbe6</span><span>gold-2:#fff1b8</span><span>gold-3:#ffe58f</span><span>gold-4:#ffd666</span><span>gold-5:#ffc53d</span><span>gold-6:#faad14</span><span>gold-7:#d48806</span><span>gold-8:#ad6800</span><span>gold-9:#874d00</span><span>gold-10:#613400</span><span>lime-1:#fcffe6</span><span>lime-2:#f4ffb8</span><span>lime-3:#eaff8f</span><span>lime-4:#d3f261</span><span>lime-5:#bae637</span><span>lime-6:#a0d911</span><span>lime-7:#7cb305</span><span>lime-8:#5b8c00</span><span>lime-9:#3f6600</span><span>lime-10:#254000</span><span>colorText:rgba(0, 0, 0, 0.88)</span><span>colorTextSecondary:rgba(0, 0, 0, 0.65)</span><span>colorTextTertiary:rgba(0, 0, 0, 0.45)</span><span>colorTextQuaternary:rgba(0, 0, 0, 0.25)</span><span>colorFill:rgba(0, 0, 0, 0.15)</span><span>colorFillSecondary:rgba(0, 0, 0, 0.06)</span><span>colorFillTertiary:rgba(0, 0, 0, 0.04)</span><span>colorFillQuaternary:rgba(0, 0, 0, 0.02)</span><span>colorBgLayout:#f5f5f5</span><span>colorBgContainer:#ffffff</span><span>colorBgElevated:#ffffff</span><span>colorBgSpotlight:rgba(0, 0, 0, 0.85)</span><span>colorBorder:#d9d9d9</span><span>colorBorderSecondary:#f0f0f0</span><span>colorPrimaryBg:#f0f9ff</span><span>colorPrimaryBgHover:#def0ff</span><span>colorPrimaryBorder:#b5dbff</span><span>colorPrimaryBorderHover:#8cc4ff</span><span>colorPrimaryHover:#63a9ff</span><span>colorPrimaryActive:#276bd9</span><span>colorPrimaryTextHover:#63a9ff</span><span>colorPrimaryText:#3a8bff</span><span>colorPrimaryTextActive:#276bd9</span><span>colorSuccessBg:#f6ffed</span><span>colorSuccessBgHover:#d9f7be</span><span>colorSuccessBorder:#b7eb8f</span><span>colorSuccessBorderHover:#95de64</span><span>colorSuccessHover:#95de64</span><span>colorSuccessActive:#389e0d</span><span>colorSuccessTextHover:#73d13d</span><span>colorSuccessText:#52c41a</span><span>colorSuccessTextActive:#389e0d</span><span>colorErrorBg:#fff2f0</span><span>colorErrorBgHover:#fff1f0</span><span>colorErrorBorder:#ffccc7</span><span>colorErrorBorderHover:#ffa39e</span><span>colorErrorHover:#ff7875</span><span>colorErrorActive:#d9363e</span><span>colorErrorTextHover:#ff7875</span><span>colorErrorText:#ff4d4f</span><span>colorErrorTextActive:#d9363e</span><span>colorWarningBg:#fffbe6</span><span>colorWarningBgHover:#fff1b8</span><span>colorWarningBorder:#ffe58f</span><span>colorWarningBorderHover:#ffd666</span><span>colorWarningHover:#ffd666</span><span>colorWarningActive:#d48806</span><span>colorWarningTextHover:#ffc53d</span><span>colorWarningText:#faad14</span><span>colorWarningTextActive:#d48806</span><span>colorInfoBg:#f0f9ff</span><span>colorInfoBgHover:#def0ff</span><span>colorInfoBorder:#b5dbff</span><span>colorInfoBorderHover:#8cc4ff</span><span>colorInfoHover:#8cc4ff</span><span>colorInfoActive:#276bd9</span><span>colorInfoTextHover:#63a9ff</span><span>colorInfoText:#3a8bff</span><span>colorInfoTextActive:#276bd9</span><span>colorBgMask:rgba(0, 0, 0, 0.45)</span><span>colorWhite:#fff</span><span>fontSizeSM:12</span><span>fontSizeLG:16</span><span>fontSizeXL:20</span><span>fontSizeHeading1:38</span><span>fontSizeHeading2:30</span><span>fontSizeHeading3:24</span><span>fontSizeHeading4:20</span><span>fontSizeHeading5:16</span><span>lineHeight:1.5714285714285714</span><span>lineHeightLG:1.5</span><span>lineHeightSM:1.6666666666666667</span><span>lineHeightHeading1:1.2105263157894737</span><span>lineHeightHeading2:1.2666666666666666</span><span>lineHeightHeading3:1.3333333333333333</span><span>lineHeightHeading4:1.4</span><span>lineHeightHeading5:1.5</span><span>sizeXXL:48</span><span>sizeXL:32</span><span>sizeLG:24</span><span>sizeMD:20</span><span>sizeMS:16</span><span>size:16</span><span>sizeSM:12</span><span>sizeXS:8</span><span>sizeXXS:4</span><span>controlHeightSM:24</span><span>controlHeightXS:16</span><span>controlHeightLG:40</span><span>motionDurationFast:0.1s</span><span>motionDurationMid:0.2s</span><span>motionDurationSlow:0.3s</span><span>lineWidthBold:2</span><span>borderRadiusXS:2</span><span>borderRadiusSM:4</span><span>borderRadiusLG:8</span><span>borderRadiusOuter:4</span><span>colorLink:#3a8bff</span><span>colorLinkHover:#8cc4ff</span><span>colorLinkActive:#276bd9</span><span>colorFillContent:rgba(0, 0, 0, 0.06)</span><span>colorFillContentHover:rgba(0, 0, 0, 0.15)</span><span>colorFillAlter:rgba(0, 0, 0, 0.02)</span><span>colorBgContainerDisabled:rgba(0, 0, 0, 0.04)</span><span>colorBorderBg:#ffffff</span><span>colorSplit:rgba(5, 5, 5, 0.06)</span><span>colorTextPlaceholder:rgba(0, 0, 0, 0.25)</span><span>colorTextDisabled:rgba(0, 0, 0, 0.25)</span><span>colorTextHeading:rgba(0, 0, 0, 0.88)</span><span>colorTextLabel:rgba(0, 0, 0, 0.65)</span><span>colorTextDescription:rgba(0, 0, 0, 0.45)</span><span>colorTextLightSolid:#fff</span><span>colorHighlight:#ff4d4f</span><span>colorBgTextHover:rgba(0, 0, 0, 0.06)</span><span>colorBgTextActive:rgba(0, 0, 0, 0.15)</span><span>colorIcon:rgba(0, 0, 0, 0.45)</span><span>colorIconHover:rgba(0, 0, 0, 0.88)</span><span>colorErrorOutline:rgba(255, 38, 5, 0.06)</span><span>colorWarningOutline:rgba(255, 215, 5, 0.1)</span><span>fontSizeIcon:12</span><span>controlOutlineWidth:2</span><span>controlInteractiveSize:16</span><span>controlItemBgHover:rgba(0, 0, 0, 0.04)</span><span>controlItemBgActive:#f0f9ff</span><span>controlItemBgActiveHover:#def0ff</span><span>controlItemBgActiveDisabled:rgba(0, 0, 0, 0.15)</span><span>controlTmpOutline:rgba(0, 0, 0, 0.02)</span><span>controlOutline:rgba(5, 155, 255, 0.06)</span><span>fontWeightStrong:600</span><span>linkDecoration:none</span><span>linkHoverDecoration:none</span><span>linkFocusDecoration:none</span><span>controlPaddingHorizontal:12</span><span>controlPaddingHorizontalSM:8</span><span>paddingXXS:4</span><span>paddingXS:8</span><span>paddingSM:12</span><span>padding:16</span><span>paddingMD:20</span><span>paddingLG:24</span><span>paddingXL:32</span><span>paddingContentHorizontalLG:24</span><span>paddingContentVerticalLG:12</span><span>paddingContentHorizontal:12</span><span>paddingContentVertical:12</span><span>paddingContentHorizontalSM:16</span><span>paddingContentVerticalSM:8</span><span>marginXXS:4</span><span>marginXS:8</span><span>marginSM:12</span><span>margin:16</span><span>marginMD:20</span><span>marginLG:24</span><span>marginXL:32</span><span>marginXXL:48</span><span>boxShadow:
      0 1px 2px 0 rgba(0, 0, 0, 0.03),
      0 1px 6px -1px rgba(0, 0, 0, 0.02),
      0 2px 4px 0 rgba(0, 0, 0, 0.02)
    </span><span>boxShadowSecondary:
      0 6px 16px 0 rgba(0, 0, 0, 0.08),
      0 3px 6px -4px rgba(0, 0, 0, 0.12),
      0 9px 28px 8px rgba(0, 0, 0, 0.05)
    </span><span>boxShadowTertiary:
      0 1px 2px 0 rgba(0, 0, 0, 0.03),
      0 1px 6px -1px rgba(0, 0, 0, 0.02),
      0 2px 4px 0 rgba(0, 0, 0, 0.02)
    </span><span>screenXS:480</span><span>screenXSMin:480</span><span>screenXSMax:575</span><span>screenSM:576</span><span>screenSMMin:576</span><span>screenSMMax:767</span><span>screenMD:768</span><span>screenMDMin:768</span><span>screenMDMax:991</span><span>screenLG:992</span><span>screenLGMin:992</span><span>screenLGMax:1199</span><span>screenXL:1200</span><span>screenXLMin:1200</span><span>screenXLMax:1599</span><span>screenXXL:1600</span><span>screenXXLMin:1600</span><span>screenXXLMax:1999</span><span>screenXXXL:2000</span><span>screenXXXLMin:2000</span><span>boxShadowPopoverArrow:3px 3px 7px rgba(0, 0, 0, 0.1)</span><span>boxShadowCard:
      0 1px 2px -2px rgba(0, 0, 0, 0.16),
      0 3px 6px 0 rgba(0, 0, 0, 0.12),
      0 5px 12px 4px rgba(0, 0, 0, 0.09)
    </span><span>boxShadowDrawerRight:
      -6px 0 16px 0 rgba(0, 0, 0, 0.08),
      -3px 0 6px -4px rgba(0, 0, 0, 0.12),
      -9px 0 28px 8px rgba(0, 0, 0, 0.05)
    </span><span>boxShadowDrawerLeft:
      6px 0 16px 0 rgba(0, 0, 0, 0.08),
      3px 0 6px -4px rgba(0, 0, 0, 0.12),
      9px 0 28px 8px rgba(0, 0, 0, 0.05)
    </span><span>boxShadowDrawerUp:
      0 6px 16px 0 rgba(0, 0, 0, 0.08),
      0 3px 6px -4px rgba(0, 0, 0, 0.12),
      0 9px 28px 8px rgba(0, 0, 0, 0.05)
    </span><span>boxShadowDrawerDown:
      0 -6px 16px 0 rgba(0, 0, 0, 0.08),
      0 -3px 6px -4px rgba(0, 0, 0, 0.12),
      0 -9px 28px 8px rgba(0, 0, 0, 0.05)
    </span><span>boxShadowTabsOverflowLeft:inset 10px 0 8px -8px rgba(0, 0, 0, 0.08)</span><span>boxShadowTabsOverflowRight:inset -10px 0 8px -8px rgba(0, 0, 0, 0.08)</span><span>boxShadowTabsOverflowTop:inset 0 10px 8px -8px rgba(0, 0, 0, 0.08)</span><span>boxShadowTabsOverflowBottom:inset 0 -10px 8px -8px rgba(0, 0, 0, 0.08)</span></div>"
`;
